ADAL/src/request/ADWebRequest.m

```diff
@@ -147,8 +148,15 @@ - (void)send
            MSID_OAUTH2_CORRELATION_ID_REQUEST_VALUE:[_correlationId UUIDString]
            }];
     }
-    NSURL* requestURL = [ADHelpers addClientVersionToURL:_requestURL];
-    requestURL = [[MSIDAadAuthorityCache sharedInstance] networkUrlForAuthority:requestURL context:self];
+    NSURL *requestURL = [ADHelpers addClientVersionToURL:_requestURL];
+    
+    __auto_type authority = [[MSIDAADAuthority alloc] initWithURL:requestURL context:self error:nil];
```

Comment:
can we skip additional checks if it's not an AAD authority? (e.g. ADFS authority)